cli/src/gdb.rs

```diff
@@ -0,0 +1,32 @@
+use std::{sync::Mutex, time::Duration};
+
+use probe_rs_cli_util::common_options::ProbeOptions;
+
+pub fn run_gdb_server(
+    common: ProbeOptions,
+    connection_string: Option<&str>,
+    reset_halt: bool,
+) -> anyhow::Result<()> {
+    let session = Mutex::new(common.simple_attach()?);
+
+    if reset_halt {
+        session
+            .lock()
+            .unwrap()
+            .core(0)?
+            .reset_and_halt(Duration::from_millis(100))?;
+    }
+
+    let gdb_connection_string = connection_string.or_else(|| Some("localhost:1337"));
```

Comment:
Could you please fix this clippy lint while we are at it even if it was not introduced now? :)